themes/default/content/blog/create-manage-confluent-kafka-cluster-with-pulumi/index.md

```diff
@@ -0,0 +1,326 @@
+1. From the command line via the `pulumi stack output` command, which you will use to test your Kafka cluster.
+
+The syntax for Pulumi stack outputs varies by language, but in TypeScript programs they are accomplished by a simple `export` statement:
+
+```typescript
+export const ordersTopicName = topic.topicName;
+export const environmentId = env.id;
+export const clusterId = cluster.id;
+
+export const producerApiKeyId = producerApiKey.id;
+export const producerApiKeySecret = producerApiKey.secret;
+
+export const consumerApiKeyId = consumerApiKey.id;
+export const consumerApiKeySecret = consumerApiKey.secret;
+```
+
+Your Pulumi program is now complete! You can deploy our infrastructure by running the following command:
+
+```bash
+pulumi up
+```
+
+After a short wait, your cluster is up and running, and you are ready to test your infrastructure!
+
+## Testing
+
+To simulate your producer and consumer, you can use the [Confluent CLI](https://docs.confluent.io/confluent-cli/current/overview.html) to send messages to and read messages from your topic. You will use the values of your Pulumi stack outputs to formulate the command.
+
+To simulate a message producer, you can send messages to your Kafka topic with the following command:
+
+```bash
+confluent kafka topic produce $(pulumi stack output ordersTopicName) \
+  --environment $(pulumi stack output environmentId) \
+  --cluster $(pulumi stack output clusterId) \
+  --api-key $(pulumi stack output producerApiKeyId) \
+  --api-secret "$(pulumi stack output producerApiKeySecret --show-secrets)"
+```
+
+You can enter a few sample records like the following and then press `Ctrl-C` when you're done to exit:
+
+```json
+{"number":1,"date":18500,"shipping_address":"899 W Evelyn Ave, Mountain View, CA 94041, USA","cost":15.00}
+{"number":2,"date":18501,"shipping_address":"1 Bedford St, London WC2E 9HG, United Kingdom","cost":5.00}
+{"number":3,"date":18502,"shipping_address":"3307 Northland Dr Suite 400, Austin, TX 78731, USA","cost":10.00}
+```
+
+To simulate your consumer and read the records you just wrote, you can enter the following command:
+
+```bash
+$ confluent kafka topic consume $(pulumi stack output ordersTopicName) \
+  --from-beginning \
+  --environment $(pulumi stack output environmentId) \
+  --cluster $(pulumi stack output clusterId) \
+  --api-key $(pulumi stack output consumerApiKeyId) \
+  --api-secret $(pulumi stack output consumerApiKeySecret --show-secrets)
+
+Starting Kafka Consumer. Use Ctrl-C to exit.
+{"number":3,"date":18502,"shipping_address":"3307 Northland Dr Suite 400, Austin, TX 78731, USA","cost":10.00}
+{"number":1,"date":18500,"shipping_address":"899 W Evelyn Ave, Mountain View, CA 94041, USA","cost":15.00}
+{"number":2,"date":18501,"shipping_address":"1 Bedford St, London WC2E 9HG, United Kingdom","cost":5.00}
```

Comment:
Would it make sense to remove the output and just copy the command? That'd make it easier to use, since the stuff below the comment isn't usable. (If you wanted to show the output, you could break the block into two parts, one to make the command copyable, the other showing the output.)

```suggestion
$ confluent kafka topic consume $(pulumi stack output ordersTopicName) \
  --from-beginning \
  --environment $(pulumi stack output environmentId) \
  --cluster $(pulumi stack output clusterId) \
  --api-key $(pulumi stack output consumerApiKeyId) \
  --api-secret $(pulumi stack output consumerApiKeySecret --show-secrets)
```

Reply:
Done.